pkg/printers/json.go

```diff
@@ -37,6 +37,13 @@ func (p *JSONPrinter) AfterPrint(w io.Writer, res string) error {
 
 // PrintObj is an implementation of ResourcePrinter.PrintObj which simply writes the object to the Writer.
 func (p *JSONPrinter) PrintObj(obj runtime.Object, w io.Writer) error {
+	gvk := obj.GetObjectKind().GroupVersionKind()
+
+	// fail on internal version given
+	if gvk.Empty() || gvk.Version == runtime.APIVersionInternal {
```

Comment:
obj.GetObjectKind().GroupVersionKind() will return an empty gvk if a versioned go struct runtime.Object is passed in (e.g. `&v1.Pod{}`)

Reply:
Thanks, updated this to only check if gvk.Version is internal